tests/pyensign/test_ensign.py

```diff
@@ -102,6 +102,39 @@ async def test_publish_exists(self, mock_publish, mock_list, ensign):
 
         errors = await ensign.publish(name, events)
         assert len(errors) == 1
+    
+    @pytest.mark.asyncio
+    @patch("pyensign.connection.Client.list_topics")
+    @patch("pyensign.connection.Client.publish")
+    async def test_publish_with_generator(self, mock_publish, mock_list, ensign):
```

Comment:
Thank you for adding this test, this is definitely a pattern I would expect users to use.